Kai/1-update-homepage-layout

CasualIntellectual opened the pull request:
I had to make a new branch as a follow up to #14 in order to give it a better name following the conventions

I have also made the requested changes to my code and renamed the pull request to better fit the content and guidelines.

Additionally, to answer your question the submission form doesn't actually work yet. I would actually like to maybe discuss with you and some other rocketry club members about what to do with it because I'm still unsure about the consensus on how submissions will. What kind of database will we use, what permissions are required, etc.

Removed text from footer
I removed the "Return to Main Page" text

Added comments to my code
I added brief comments to my code

Added blank lines to all files
Added newline at end of files

Added loops and reduced repetitive code
Remove code blocks using loops

Abstracted header and footer to another file using jQuery
Header and footer are now separate files

Removed text file

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/waterloo-rocketry/parts.website/15)
<!-- Reviewable:end -->

CasualIntellectual committed:
Update README.md

CasualIntellectual committed:
Add files via upload

CasualIntellectual committed:
Create image.txt

CasualIntellectual committed:
Add files via upload

CasualIntellectual committed:
Add files via upload

Made New Branch Following Naming Rules, Addressed Changes

CasualIntellectual committed:
Delete image/image.txt

Excelsium2 commented:
I broke the dropdown button though so I need to work on that

CasualIntellectual committed:
Add files via upload

Fixed navbar, remove submit page for now

Excelsium2 commented:
Update:

Fixed navbar and removed submit page since it is not functional right now

CasualIntellectual committed:
Update about.html

CasualIntellectual committed:
Update footer.html

CasualIntellectual committed:
Update general.css

CasualIntellectual committed:
Add files via upload

CasualIntellectual committed:
Add files via upload

CasualIntellectual committed:
Add files via upload

Darth-Raazi reviewed (changes requested):
Great work on this Kai! I especially really like how you handled resizing on small screens. Left a couple small comments. Also, I think the footer positioning on the about page needs some tweaking:
![image.png](https://files.reviewableusercontent.io/images/reviewable/67076803/3AMPiQlvPtTfghB8_1Ea90o5/image.png)

> Reviewed 17 of 20 files at r1, 2 of 3 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/parts.website/15#-:-NuPpGjk2IBbBDn3yEQU:bhxte37)* status: all files reviewed, 2 unresolved discussions (waiting on @CasualIntellectual)

___
*[`general.css` line 385 at r3](https://reviewable.io/reviews/waterloo-rocketry/parts.website/15#-NuPnij5Amz7J7hl0QMR:-NuPnij5Amz7J7hl0QMS:b-50ih20) ([raw file](https://github.com/waterloo-rocketry/parts.website/blob/85ae790a9007011a7e8b30acdf047b712739d2fc/general.css#L385)):*
> ```CSS
> }
> 
> /* ====================  RESIZING  ================== */
> ```

Stale comment or did you forget something here?
___
*[`sheets.js` line 47 at r3](https://reviewable.io/reviews/waterloo-rocketry/parts.website/15#-NuPnuqa11OFVZVLNlyC:-NuPnuqa11OFVZVLNlyD:b-exxgnd) ([raw file](https://github.com/waterloo-rocketry/parts.website/blob/85ae790a9007011a7e8b30acdf047b712739d2fc/sheets.js#L47)):*
> ```JavaScript
>     return gapi.client.sheets.spreadsheets.values.get({
>         spreadsheetId: SHEET_ID,
>         range: 'A:H',
> ```

Why are we only loading select columns?


<!-- Sent from Reviewable.io -->

CasualIntellectual committed:
Add files via upload

-Modified footer on "About" page
-Added additional collumns from sheets (it doesn't seem like I can take the full range of collumns, but maybe I'm not using the API correctly)

Excelsium2 commented:
Now loading all collumns in google sheets

Excelsium2 commented:
Also adjusted footer

Excelsium2 commented:
As far as I know there doesn't seem to be a way to access all collumns automatically for the website's purposes.

Darth-Raazi reviewed (changes requested):
Good work, couple things:
- There is a certain page width where the footer appears and permanently stays there, even if you're not at the bottom of the page
- Both the header and the footer (but especially the footer), look a tad tall to me. Play around with the height and see what looks nice
- Both the header and footer are a bit transparent so when you scroll down and some of the content goes behind the header, you can still see it and it looks bad. Both should be solid

> Reviewed 5 of 5 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/parts.website/15#-:-NzaHVeG4BDsKJ5nb2Mt:b-fip026)* status: all files reviewed, 4 unresolved discussions (waiting on @CasualIntellectual)

___
*[`general.css` line 391 at r4](https://reviewable.io/reviews/waterloo-rocketry/parts.website/15#-NzaGClw6dsCJ3b9RVos:-NzaGClw6dsCJ3b9RVot:b-evif80) ([raw file](https://github.com/waterloo-rocketry/parts.website/blob/d5d69d192edceecf0590227d1570ed0d058011e8/general.css#L391)):*
> ```CSS
>         position: fixed;
>     }
> }
> ```

Nit: newline please
___
*[`table.js` line 48 at r4](https://reviewable.io/reviews/waterloo-rocketry/parts.website/15#-NzaG_-04zYYN7uxSHSZ:-NzaG_-04zYYN7uxSHS_:bea6t9u) ([raw file](https://github.com/waterloo-rocketry/parts.website/blob/d5d69d192edceecf0590227d1570ed0d058011e8/table.js#L48)):*
> ```JavaScript
>         );
>         document.getElementById("select1").src = "image/selected.png"
>         document.getElementById("select3").style.display = "none";
> ```

What was `select2` and why did it get removed from here?


<!-- Sent from Reviewable.io -->

anubhavsengupta committed:
Style updates

anubhavsengupta committed:
Grouped CSS into different files

anubhavsengupta reviewed (commented):
1. Fixed footer appearing and permenanently staying at position.

2. Reduced with of header and footer

3. Removed transparency

I also made a styles folder and created a few CSS files for organization. I made minimal changes to some of the table-rendering logic when the screen is resized.

> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/parts.website/15#-:-O1UJAad7ZFz4izL6KIv:b8x0rsi)* status: 14 of 23 files reviewed, 4 unresolved discussions (waiting on @CasualIntellectual and @Darth-Raazi)



<!-- Sent from Reviewable.io -->

anubhavsengupta committed:
Remove redundant stylesheet

anubhavsengupta committed:
Clean up comments and repetitive styles
